modules/oe_translation_poetry/tests/Functional/PoetryTranslationLanguageAddTest.php

```diff
@@ -104,7 +104,130 @@ public function testTranslationAddLanguagesRequest(): void {
     $this->drupalGet($node->toUrl('drupal:content-translation-overview'));
 
     // Assert button we see the update and add language buttons again.
-    $this->assertSession()->buttonExists('Add extra languages to the DGT request');
+    $this->assertSession()->buttonExists('Add extra languages to the ongoing DGT request');
+    $this->assertSession()->buttonExists('Request a DGT translation update for the selected languages');
+    $this->assertSession()->pageTextNotContains('No translation requests to DGT can be made until the ongoing ones have been accepted and/or translated.');
+
+    // Send the translations for each job.
+    $this->notifyWithDummyTranslations($jobs);
+    $this->assertJobsAreTranslated();
+    foreach ($this->loadJobsKeyedByLanguage() as $job) {
+      $this->assertEquals(PoetryTranslator::POETRY_STATUS_TRANSLATED, $job->get('poetry_state')->value);
+    }
+  }
+
+  /**
+   * Tests to add a language to a request after content has been updated.
+   */
+  public function testTranslationAddLanguagesRequestAfterUpdate(): void {
+    $node = $this->createNodeWithRequestedJobs([
+      'title' => 'My node title',
+      'field_oe_demo_translatable_body' => 'My node body',
+    ], ['bg', 'cs']);
+
+    // Keep track of the original node revision ID because it is the one that
+    // should get saved on the job item.
+    $revision_id = $node->getRevisionId();
+
+    // Assert the newly created job items have the correct node revision ID and
+    // data.
+    $jobs = $this->loadJobsKeyedByLanguage();
+    foreach ($jobs as $job) {
+      $job_items = $job->getItems();
+      foreach ($job_items as $job_item) {
+        $this->assertEquals($revision_id, $job_item->get('item_rid')->value);
+        $this->assertEquals($node->bundle(), $job_item->get('item_bundle')->value);
+        $data = $job_item->getData();
+        $this->assertEquals('My node title', $data['title'][0]['value']['#text']);
+      }
+    }
+
+    $this->drupalGet($node->toUrl('drupal:content-translation-overview'));
```

Comment:
I don't think we need to navigate to the translation overview at this point (we do it again later anyways)